chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -785,19 +800,31 @@ func (b *BitcoindNotifier) historicalSpendDetails(
 	spendRequest chainntnfs.SpendRequest, startHeight, endHeight uint32) (
 	*chainntnfs.SpendDetail, error) {
 
-	// Begin scanning blocks at every height to determine if the outpoint
-	// was spent.
+	// batchRequest will store the scan requests at every height to determine
+	// if the output was spent.
+	batchRequest := make(map[uint32]rpcclient.FutureGetBlockHashResult, endHeight-startHeight)
 	for height := endHeight; height >= startHeight && height > 0; height-- {
-		// Ensure we haven't been requested to shut down before
-		// processing the next height.
+
+		// BatchClient works with AsynRequest.
+		sendReq, err := b.chainConn.GetBlockHashAsync(int64(height))
+		if err != nil {
+			return nil, fmt.Errorf("unable to receive the block promise %v", err)
+		}
+		batchRequest[height] = sendReq
+	}
+
+	// Sending the bulk request using updated batchClient.
+	b.chainConn.SendAsyncQueue()
+
+	for height, response := range batchRequest {
```

Comment:
Again, this loop could potentially be refactored to fetch the blocks async as well.